mirascope/base/extractors.py

```diff
@@ -125,22 +144,34 @@ def _extract(
         kwargs, return_tool = self._setup(tool_type, kwargs)
 
         class TempCall(call_type):  # type: ignore
-            prompt_template = self.prompt_template
+            prompt_template = """
+            {original_prompt_template}
+
+            {error_message}
+            """
 
             base_url = self.base_url
             api_key = self.api_key
             call_params = self.call_params
-
+            error_message: str = ""
+            original_prompt_template: str = self.prompt_template
             model_config = ConfigDict(extra="allow")
 
         properties = getmembers(self)
         for name, value in properties:
             if not hasattr(TempCall, name):
                 setattr(TempCall, name, value)
-
-        response = TempCall(**self.model_dump(exclude={"extract_schema"})).call(
-            **kwargs
-        )
+        error_message = kwargs.pop("error_message", "")
+        if error_message and error_message == "No tool found in the completion":
```

Comment:
this is unnecessary, just pass the error message in and tell it to fix it (it should identify that "no tool" and "please fix" means "use tool")